### ISSUE TITLE
[Bug]: SMTP not working

### ISSUE THREAD

**Issue by dbergloev** [nextcloud/server#42946]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Hi

I have Protonmail Bridge setup on a server that is also running NextCloud. I tried setting up SMTP on Nextcloud to allow for emails to be sent, like resetting passwords and such, but I could not get this working. I also cannot find any logs anywhere to help, even though I have tried enabling debug for email. I have seached this issue all day, and have applied a few possible fixes in regards to self signed certificates and such, but this also did not fix anything. It does seam to connect to the SMTP service. If I close off the port, it fails right away. If I open up the port again, it loads for awhile before erroring out. 

I have manually checked the SMTP connection using openssl and it works fine. I also tried sending an email through this connection and that also worked, so there should not be any issue with the bridge.

```
$ openssl s_client -starttls smtp -connect 172.22.0.188:25 -crlf -ign_eof
CONNECTED(00000003)
Can't use SSL_get_servername
depth=0 C = CH, O = Proton AG, OU = Proton Mail, CN = 127.0.0.1
verify error:num=18:self-signed certificate
verify return:1
depth=0 C = CH, O = Proton AG, OU = Proton Mail, CN = 127.0.0.1
verify return:1
---
Certificate chain
 0 s:C = CH, O = Proton AG, OU = Proton Mail, CN = 127.0.0.1
   i:C = CH, O = Proton AG, OU = Proton Mail, CN = 127.0.0.1
   a:PKEY: rsaEncryption, 2048 (bit); sigalg: RSA-SHA256
   v:NotBefore: Jan 18 17:41:40 2024 GMT; NotAfter: Jan 13 17:41:40 2044 GMT
---
Server certificate
-----BEGIN CERTIFICATE-----
MIIDkjCCAnqgAwIBAgIQA4ADvxfk3gR510ZvcqwAbDANBgkqhkiG9w0BAQsFADBL
b24gTWFpbDESMBAGA1UEAxMJMTI3LjAuMC4xMB4XDTI0MDExODE3NDE0MFoXDTQ0
FdEWqRsDwV6usXoYyzOWtmtk6FbQCVQ5LJgKsDk7dz7aQZ0XeOwg/ZEtQo5p/cOP
MDExMzE3NDE0MFowSzELMAkGA1UEBhMCQ0gxEjAQBgNVBAoTCVByb3RvbiBBRzEU
MBIGA1UECxMLUHJvdG9uIE1haWwxEjAQBgNVBAMTCTEyNy4wLjAuMTCCASIwDQYJ
KoZIhvcNAQEBBQADggEPADCCAQoCggEBAK+0BjvbZiTImU9L1NXVN3vl9PqTLJcy
s/RR2108j6WAFnDeriKmJcChBgZ84DXrR/1QDE/Ak6W8K9/nLbHZc1Vw3m5i/rQB
qq+VvGf3xPQVe0o8RJOsFjx5S8ORFQjS+Zjrc7GuCVqdLALBj0OZNtnzca/cphmS
k9GI5OGETg1l6r2sHLJ2mhsZac0/duio9PCkJfezvTFco/mtp9DO1FgtYAgy9GU0
4Iri3EFCL6Lq10tieff0az6CikzM6MEIHds2HOs/F3fZVTYArSbdl2MCAwEAAaNy
MHAwDgYDVR0PAQH/BAQDAgKkMB0GA1UdJQQWMBQGCCsGAQUFBwMBBggrBgEFBQcD
AjAPBgNVHRMBAf8EBTADAQH/MB0GA1UdDgQWBBQdECgrjda91F9MFwOd6iQ195vS
fTAPBgNVHREECDAGhwR/AAABMA0GCSqGSIb3DQEBCwUAA4IBAQBDVAkdOAdPyR2+
BE5pLtepgjZHluSJHO4d/P4dC/EQKY14BkA4Z9qQRDnADl1pl9reP9xEA4T1BU60
qHIoMbPQP2dBKwj/jnnp6/JiOoTUaRP8Lw3Yj3XY15EYxLd2Ych83XWQIZ+1Q1oZ
PDdtucoK3al3DxKsEWYB4H4RuAN1cs68MH6ZxnRo0qZLKFJnUf0kvQ0yhAry+k1N
SflTfJteHpJLgYQRA6Is7Ordxt7/ezwjY+lpNCBm6AeJRHHXRpE9I3L8v/7chcXc
MQswCQYDVQQGEwJDSDESMBAGA1UEChMJUHJvdG9uIEFHMRQwEgYDVQQLEwtQcm90
9kzq8fPdnK6lPs8/XjN7/O0XVHC1ZFXkQ4FG5to2QMTv2xeG1nGjtUuPOC/AuK2a
gsfaExmn
-----END CERTIFICATE-----
subject=C = CH, O = Proton AG, OU = Proton Mail, CN = 127.0.0.1
issuer=C = CH, O = Proton AG, OU = Proton Mail, CN = 127.0.0.1
---
No client certificate CA names sent
Peer signing digest: SHA256
Peer signature type: RSA-PSS
Server Temp Key: X25519, 253 bits
---
SSL handshake has read 1666 bytes and written 394 bytes
Verification error: self-signed certificate
---
New, TLSv1.3, Cipher is TLS_AES_128_GCM_SHA256
Server public key is 2048 bit
Secure Renegotiation IS NOT supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
Early data was not sent
Verify return code: 18 (self-signed certificate)
---
250 SIZE
---
Post-Handshake New Session Ticket arrived:
SSL-Session:
    Protocol  : TLSv1.3
    Cipher    : TLS_AES_128_GCM_SHA256
    Session-ID: 292EDDCF9B57A0ACBF9ADFD84400571F4709EFB8ECD4CBC2EA9F1BFBF4836B98
    Session-ID-ctx: 
    Resumption PSK: E8E36F590E3FBBC8859B23CF6A5BDA70C8E66810988DBECFB3C3D783468705B9
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    TLS session ticket lifetime hint: 604800 (seconds)
    TLS session ticket:
    0000 - 67 43 db 50 7f 08 7b 6c-61 86 2d 65 1b f7 b1 0b   gC.P..{la.-e....
    0010 - 1d aa fc 5f 33 55 8a bf-4d e3 45 f4 3e fe ab 04   ..._3U..M.E.>...
    0020 - 65 28 26 f6 a5 d3 a8 0c-9e 2e 67 aa f1 78 a8 3a   e(&.......g..x.:
    0030 - ea fc ff c3 db 45 73 a8-59 76 9c 46 bc dc 97 d4   .....Es.Yv.F....
    0040 - ff 8c b9 4a ed 86 92 c5-9f 1c 89 58 29 f2 6c c0   ...J.......X).l.
    0050 - 16 76 34 fb de c9 a4 29-62 44 d5 b3 33 ac a4 6c   .v4....)bD..3..l
    0060 - 80 8f 87 e8 98 6d 75 c2-db 22 99 6c 12 2c 07 ab   .....mu..".l.,..
    0070 - e0                                                .

    Start Time: 1705610896
    Timeout   : 7200 (sec)
    Verify return code: 18 (self-signed certificate)
    Extended master secret: no
    Max Early Data: 0
---
read R BLOCK
EHLO localhost
250-Hello localhost
250-PIPELINING
250-8BITMIME
250-ENHANCEDSTATUSCODES
250-CHUNKING
250-AUTH PLAIN LOGIN
250 SIZE
AUTH PLAIN AHVzZXIxQGRvbWFpbgB3N0s5VE5NM1RfLXVyc2R2c2R2c2R2Z1hkbFE=
235 2.0.0 Authentication succeeded
```

```
AUTH PLAIN AHVzZXIxQGRvbWFpbgB3N0s5VE5NM1RfLXVyc2R2c2R2c2R2Z1hkbFE=
235 2.0.0 Authentication succeeded
MAIL FROM: <user1@domain>
250 2.0.0 Roger, accepting mail from <user1@domain>
rcpt to: <user2@domain>
250 2.0.0 I'll make sure <user2@domain>> gets this
DATA
354 2.0.0 Go ahead. End your data with <CR><LF>.<CR><LF>
Subject: It worked

Hi there!
.
250 2.0.0 OK: queued
quit
221 2.0.0 Bye
closed
```

### Steps to reproduce

1. Install NextCloud
2. Install Protonmail Bridge
3. Try to send an email via Basic Settings

### Expected behavior

That emails is sent 

### Installation method

Community Docker image

### Nextcloud Server version

28

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.2

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Fresh Nextcloud Server install

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "htaccess.RewriteBase": "\/",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "apps_paths": [
            {
                "path": "\/var\/www\/html\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/www\/html\/custom_apps",
                "url": "\/custom_apps",
                "writable": true
            }
        ],
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "nextcloud.domain"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "28.0.1.1",
        "overwrite.cli.url": "https:\/\/nextcloud.domain",
        "overwriteprotocol": "https",
        "overwritehost": "nextcloud.domain",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "logtimezone": "Europe\/Copenhagen",
        "default_phone_region": "DK",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_sendmailmode": "smtp",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "25",
        "mail_smtpauth": 1,
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpsecure": "",
        "mail_smtpdebug": true,
        "mail_smtpstreamoptions": {
            "ssl": {
                "allow_self_signed": true,
                "verify_peer": false,
                "verify_peer_name": false
            }
        },
        "theme": "",
        "loglevel": 2,
        "maintenance": false
    }
}
```

### List of activated Apps

```shell
Enabled:
  - activity: 2.20.0
  - circles: 28.0.0-dev
  - cloud_federation_api: 1.11.0
  - comments: 1.18.0
  - contactsinteraction: 1.9.0
  - dav: 1.29.1
  - federatedfilesharing: 1.18.0
  - federation: 1.18.0
  - files: 2.0.0
  - files_pdfviewer: 2.9.0
  - files_reminders: 1.1.0
  - files_sharing: 1.20.0
  - files_trashbin: 1.18.0
  - files_versions: 1.21.0
  - firstrunwizard: 2.17.0
  - logreader: 2.13.0
  - lookup_server_connector: 1.16.0
  - nextcloud_announcements: 1.17.0
  - notifications: 2.16.0
  - oauth2: 1.16.3
  - password_policy: 1.18.0
  - photos: 2.4.0
  - privacy: 1.12.0
  - provisioning_api: 1.18.0
  - recommendations: 2.0.0
  - related_resources: 1.3.0
  - serverinfo: 1.18.0
  - settings: 1.10.1
  - sharebymail: 1.18.0
  - support: 1.11.0
  - survey_client: 1.16.0
  - systemtags: 1.18.0
  - text: 3.9.1
  - theming: 2.3.0
  - twofactor_backupcodes: 1.17.0
  - updatenotification: 1.18.0
  - user_status: 1.8.1
  - viewer: 2.2.0
  - weather_status: 1.8.0
  - workflowengine: 2.10.0
Disabled:
  - admin_audit: 1.18.0
  - bruteforcesettings: 2.8.0
  - dashboard: 7.8.0 (installed 7.7.0)
  - encryption: 2.16.0
  - files_external: 1.20.0
  - files_rightclick: 0.15.1 (installed 1.6.0)
  - suspicious_login: 6.0.0
  - twofactor_totp: 10.0.0-beta.2
  - user_ldap: 1.19.0
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Nextcloud Logs

```shell
Empty
```

**Comment by joshtrichards** [comment]
For `mail_smtpdebug` to work you also need to turn on Nextcloud `debug`:

https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/email_configuration.html#enabling-debug-mode

**Comment by dbergloev** [comment]
Great. I just love issues that corrects themselves without knowing why. I added the debug level and set it to 0, looking forward to see an actual log of this issue and the mail just got sent without any problems. I removed the debug level (had to try) and it still works. 

I spent most of the day yesterday trying 1000 things. Maybe the last try was the fix, but did not get loaded from the config file when I testet it. Can't think of much else since I refuse to think that the debug level some how magically fixed something.
